app/models/user.rb

```diff
@@ -140,7 +155,15 @@ def update_access_token
 
   private
 
+  def oauth_provided?
+    provider.present? || token.present?
+  end
+
   def google_access_token_expired?
     !access_expires_at || Time.zone.now > access_expires_at
   end
+
+  def facebook_access_token_expired?
```

Comment:
Same comment here abstract `google_access_token_expired?` and `facebook_access_token_expired?` into one method